SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp

```diff
@@ -41,30 +53,75 @@ void USpatialDispatcher::ProcessOps(Worker_OpList* OpList)
 
 		// Components
 		case WORKER_OP_TYPE_ADD_COMPONENT:
-			StaticComponentView->OnAddComponent(Op->add_component);
-			Receiver->OnAddComponent(Op->add_component);
+			if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(Op->add_component.data.component_id))
+			{
+				UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+				UserCallback->OnAddComponent(Op->add_component);
+			}
+			else
+			{
+				StaticComponentView->OnAddComponent(Op->add_component);
+				Receiver->OnAddComponent(Op->add_component);
+			}
 			break;
+
 		case WORKER_OP_TYPE_REMOVE_COMPONENT:
+			if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(Op->remove_component.component_id))
+			{
+				UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+				UserCallback->OnRemoveComponent(Op->remove_component);
+			}
 			break;
 		case WORKER_OP_TYPE_COMPONENT_UPDATE:
-			QueuedComponentUpdateOps.Add(Op);
-			StaticComponentView->OnComponentUpdate(Op->component_update);
+			if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(Op->component_update.update.component_id))
+			{
+				UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+				UserCallback->OnComponentUpdate(Op->component_update);
+			}
+			else
+			{
+				QueuedComponentUpdateOps.Add(Op);
+				StaticComponentView->OnComponentUpdate(Op->component_update);
+			}
 			break;
 
 		// Commands
 		case WORKER_OP_TYPE_COMMAND_REQUEST:
-			Receiver->OnCommandRequest(Op->command_request);
+			if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(Op->command_request.request.component_id))
+			{
+				UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+				UserCallback->OnCommandRequest(Op->command_request);
+			}
+			else
+			{
+				Receiver->OnCommandRequest(Op->command_request);
+			}
 			break;
 		case WORKER_OP_TYPE_COMMAND_RESPONSE:
-			Receiver->OnCommandResponse(Op->command_response);
+			if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(Op->command_response.response.component_id))
+			{
+				UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+				UserCallback->OnCommandResponse(Op->command_response);
+			}
+			else
+			{
+				Receiver->OnCommandResponse(Op->command_response);
+			}
 			break;
 
 		// Authority Change
 		case WORKER_OP_TYPE_AUTHORITY_CHANGE:
-			StaticComponentView->OnAuthorityChange(Op->authority_change);
-			Receiver->OnAuthorityChange(Op->authority_change);
+			if (UOpCallbackTemplate** UserCallbackWrapper = UserOpCallbacks.Find(Op->authority_change.component_id))
+			{
+				UOpCallbackTemplate* UserCallback = *UserCallbackWrapper;
+				UserCallback->OnAuthorityChange(Op->authority_change);
+			}
+			else
+			{
+				StaticComponentView->OnAuthorityChange(Op->authority_change);
```

Comment:
Any reason we don't update the view with these components? Seems like it would be useful.

Reply:
Would you expect the behaviour to be all external components (IDs from 1000 to 2000) to be registered in the view or just the components for which the user has registered callbacks? Either way, compared to the existing implementation of `USpatialStaticComponentView::OnAddComponent` with a switch on known component IDs, I can imagine something which takes the external schema component ID and data and just calls `Data = MakeUnique<improbable::ComponentStorage<Component>>(Op.data);`- using the base class as opposed to the handcrafted derived classes of this for each std component?

Reply:
added auth changes to static component view auth map 👍